### DISCUSSION TITLE
Welcome to SimpleIdServer Discussions!

### DISCUSSION THREAD

**Discussion by simpleidserver** [simpleidserver/SimpleIdServer#190] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Wavesolid** [comment]
Hello guys,

It seems your documentation about creating a new template IdServer is obsolete. Because `--connectionString` is no longer a valid argument.

**Comment by simpleidserver** [comment]
Hello,

Thank you for your remark :)
I create a ticket to update the documentation about using the template to create a SimpleIdServer solution. https://github.com/simpleidserver/SimpleIdServer/issues/555

**Comment by LAKSHMAN-M** [comment]
what are the user and groups default endpoints to test the SCIM application?
I tried using the localhost:5003/users as per mentioned in the document https://simpleidserver.com/docs/scim20#post-user ?
Can anyone reply quick if noticed? It is needed for the current project...

**Reply by simpleidserver** [reply]
Hello,

You can access all the operations available in the SCIM project through its Swagger endpoint:
http://localhost:5003/swagger/index.html

User endpoint: https://localhost:5003/Users
Group endpoint: https://localhost:5003/Groups

KR,
SID

**Comment by LAKSHMAN-M** [comment]
Hello Sid,

Thank you for your quick response. When I opened the link http://localhost:5003/swagger/index.html, I could see the users, groups, and other endpoints. However, I tried creating some users, and it resulted in a 401 error code.

Could you let me know if a token needs to be passed for this operation? Also, can you explain the meaning of the prefix in the URL with a small example?

<img width="717" alt="Screenshot 2024-12-22 114123" src="https://github.com/user-attachments/assets/ad30f38c-c7fd-40a0-8066-7d05a21d4ba8" />
Looking forward to your guidance.

Best regards,
Lakshman

**Comment by thabart** [comment]
Hello,

An unauthorized error is returned because you must pass a valid token in the Authorization header. For example:
Authorization: Bearer ba521b3b-02f7-4a37-b03c-58f713bf88e7

You can find other possible values in the documentation: https://simpleidserver.com/docs/scim20#security.

By default, the prefix is not used. This parameter is required when the realm is enabled in the project.

________________________________
De : Lakshman M ***@***.***>
Envoyé : dimanche 22 décembre 2024 07:16
À : simpleidserver/SimpleIdServer ***@***.***>
Cc : Subscribed ***@***.***>
Objet : Re: [simpleidserver/SimpleIdServer] Welcome to SimpleIdServer Discussions! (Discussion #190)


Hello Sid,

Thank you for your quick response. When I opened the link http://localhost:5003/swagger/index.html, I could see the users, groups, and other endpoints. However, I tried creating some users, and it resulted in a 401 error code.

Could you let me know if a token needs to be passed for this operation? Also, can you explain the meaning of the prefix in the URL with a small example?

Screenshot.2024-12-22.114123.png (view on web)<https://github.com/user-attachments/assets/ad30f38c-c7fd-40a0-8066-7d05a21d4ba8> Looking forward to your guidance.

Best regards,
Lakshman

**Comment by LAKSHMAN-M** [comment]
Hi SID,
![image](https://github.com/user-attachments/assets/53471a77-d60a-4e02-bcff-401085516bea)
I added the primary in the addresses section but it still shows an same (SimpleIdServer.Scim.Exceptions.SCIMSchemaViolatedException: attribute primary is not recognized by the SCIM schema) error. How to resolve this?
your guidance.
Best regards,
Lakshman

**Reply by simpleidserver** [reply]
Hello,

First, I apologize for the delayed response.
When the UserSchema.json file is updated after the database and its tables have already been migrated, the changes will not be automatically applied :(.
As a result, the primary attribute is not recognized by the SCIM server.

To resolve this issue, you can perform one of the following actions:

* Drop the SCIM database and restart the application.
* Execute an SQL query to manually insert the missing record into the scim.SCIMSchemaAttribute table.

![image](https://github.com/user-attachments/assets/69069c45-63a3-4a6e-b17d-4805a075fca6)

KR,
SID
